include/CXXGraph/Graph/Algorithm/Krager_impl.hpp

```diff
@@ -0,0 +1,66 @@
+#include <iostream>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+#include <random>
+#include "Node_impl.h"
+#include "CXXGraph/Graph/Graph_decl.h"
+
+
+namespace CXXGraph {
+
+template <typename T>
+class Graph {
+private:
+    std::unordered_map<CXXGraph::id_t, Node<T>> nodes;
```

Comment:
in `Graph_decl.h` we already have the node set and the adjiacency matrix, you should use them